src/parameters/parameters_flow_solver.cpp

```diff
@@ -116,6 +116,42 @@ void FlowSolverParam::declare_parameters(dealii::ParameterHandler &prm)
                               dealii::Patterns::Bool(),
                               "Use the input .msh file which calls read_gmsh. False by default.");
 
+            prm.declare_entry("use_periodic_BC_in_x", "false",
+                              dealii::Patterns::Bool(),
+                              "Use periodic boundary condition in the x-direction. False by default.");
+
+            prm.declare_entry("use_periodic_BC_in_y", "false",
+                              dealii::Patterns::Bool(),
+                              "Use periodic boundary condition in the y-direction. False by default.");
+
+            prm.declare_entry("use_periodic_BC_in_z", "false",
+                              dealii::Patterns::Bool(),
+                              "Use periodic boundary condition in the z-direction. False by default.");
+
+            prm.declare_entry("x_periodic_id_face_1", "2001",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the first periodic boundary face in the x-direction.");
+
+            prm.declare_entry("x_periodic_id_face_2", "2002",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the second periodic boundary face in the x-direction.");
+
+            prm.declare_entry("y_periodic_id_face_1", "2003",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the first periodic boundary face in the y-direction.");
+
+            prm.declare_entry("y_periodic_id_face_2", "2004",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the second periodic boundary face in the y-direction.");
+
+            prm.declare_entry("z_periodic_id_face_1", "2005",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the first periodic boundary face in the z-direction.");
+
+            prm.declare_entry("z_periodic_id_face_2", "2006",
+                              dealii::Patterns::Integer(1, dealii::Patterns::Integer::max_int_value),
+                              "Boundary ID for the second periodic boundary face in the z-direction.");
+
```

Comment:
put these in a subsection within grid called "boundary_IDs"

Reply:
Fixed this in the recent commit.